### DISCUSSION TITLE
$wire is undefined

### DISCUSSION THREAD

**Discussion by jordyvandomselaar** [livewire/livewire#2034] [1. Help]
I have this PHP component:

```php
<?php

declare(strict_types=1);

namespace App\View\Components\Livewire;
use Illuminate\Contracts\View\View as ViewInterface;
use Livewire\Component;

class RatingBar extends Component
{
    public string $foo = 'test123';

    public function render(): ViewInterface
    {
        return view('components.livewire.rating-bar.rating-bar');
    }
}
```

With this view:

```html
<div :data-foo="$wire.foo"></div>
```

and this Layout:

```html
<!doctype html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}" prefix="og: https://ogp.me/ns#">
<head>
    <livewire:styles />
</head>
    <body>
        <main class="main {{ $classes ?? '' }}">
            @yield('content')
        </main>
        <livewire:scripts />
    </body>
</html>
```

`dataSet.foo` is undefined. Is there something I'm missing?

**Comment by joshhanley** [comment]
Because you are using @yield in your layout file then you will need to add `->extends('layouts.app')` (or whatever your layout file is called) into your render method. Check it out here https://laravel-livewire.com/docs/2.x/rendering-components#custom-layout

If you don't want to do that, I'd recommend changing over to using blade component `{{ $slot }}` in your layout file

Hope this helps!

**Reply by jordyvandomselaar** [reply]
The components are rendered inside blade files that do extend the layout, sorry that I forgot to mention that

**Reply by joshhanley** [reply]
Ah ok cool. Sorry I just realised I didn't look closely at the title. $wire is an AlpineJs helper. Try @this.foo instead
